safety_shield/include/safety_shield/safety_shield.h

```diff
@@ -253,46 +273,84 @@ class SafetyShield {
 
   //////// Reachable sets of human and robot //////
   /**
-   * @brief Vector of robot reachable set capsules (get updated in every step()).
+   * @brief Vector of robot reachable set capsules for full-stop criterion (get updated in every step()).
    */
   std::vector<reach_lib::Capsule> robot_capsules_;
 
   /**
-   * @brief Vector of human reachable set capsules (get updated in every step()).
+   * @brief Vector of human reachable set capsules for full-stop criterion (get updated in every step()).
    */
   std::vector<std::vector<reach_lib::Capsule>> human_capsules_;
 
+  /**
+   * @brief Vector of robot reachable set capsules for velocity criterion (get updated in every step()).
+   */
+  std::vector<reach_lib::Capsule> robot_capsules_velocity_;
```

Comment:
we can delete the four extra capsules, because we use the two standard ones (from above)